downloader.go

```diff
@@ -240,41 +239,53 @@ func (d *Downloader) prepareAndStartDownload(ctx context.Context, url string, wg
 		ch <- s
 		return
 	}
-	defer f.Close()
 	if err := f.Truncate(int64(t)); err != nil {
 		s.Error = fmt.Errorf("error truncating %s to %d: %w", path, t, err)
 		ch <- s
 		return
 	}
+	var urlDownload sync.WaitGroup
+	defer func() {
+		urlDownload.Wait()
+		f.Close()
+	}()
```

Comment:
Shouldn't we move that to line 242? So if `f.Truncate(…)` errors, we still `f.Close()`.

Reply:
Great point. Done.